Library/Homebrew/formula.rb

```diff
@@ -1008,7 +1008,7 @@ def caveats
   # rarely, you don't want your library symlinked into the main prefix
   # see gettext.rb for an example
   def keg_only?
-    keg_only_reason && keg_only_reason.valid?
+    !!keg_only_reason && keg_only_reason.valid?
```

Comment:
If I specify `keg_only`, wouldn‘t it still call https://github.com/Homebrew/brew/blob/fd5e673d2d367eff03f1411d909ca0bbb2003a2c/Library/Homebrew/formula.rb#L2343?